### PR TITLE
ARROW-383: [C++] Integration testing CLI tool

### PR DESCRIPTION
**Pull request by wesm**
Modeled after Java version in ARROW-367

### CODE REVIEW

**Comment by wesm** [comment]
Don't know what's going on with Travis CI, but this is ready to do cc @xhochy

**Comment by wesm** [comment]
Java failed due to a transient apt-get issue. I'm going to merge this so I can proceed with ARROW-363 -- if there are comments on this patch I can address them in #211

**Comment by wesm** [comment]
+1

File: cpp/src/arrow/ipc/CMakeLists.txt [hunk 1]
```diff
@@ -85,6 +85,33 @@ ADD_ARROW_TEST(ipc-json-test)
 ARROW_TEST_LINK_LIBRARIES(ipc-json-test
   ${ARROW_IPC_TEST_LINK_LIBS})
 
+ADD_ARROW_TEST(json-integration-test)
+
+if (APPLE)
+  target_link_libraries(json-integration-test
+    arrow_static
```
Comment (xhochy) [inline cpp/src/arrow/ipc/CMakeLists.txt]: Would be nice to factor the common libraries out and only list the apple-specific ones here.

**Pull request closed without merging**